cpmpy/solvers/cpo.py

```diff
@@ -0,0 +1,491 @@
+import shutil
+
+from .solver_interface import SolverInterface, SolverStatus, ExitStatus
+from .. import DirectConstraint
+from ..expressions.core import Expression, Comparison, Operator, BoolVal
+from ..expressions.globalconstraints import GlobalConstraint
+from ..expressions.globalfunctions import GlobalFunction
+from ..expressions.variables import _BoolVarImpl, NegBoolView, _IntVarImpl, _NumVarImpl
+from ..expressions.utils import is_num, is_any_list, eval_comparison, argval, argvals
+from ..transformations.get_variables import get_variables
+from ..transformations.normalize import toplevel_list
+from ..transformations.decompose_global import decompose_in_tree
+
+"""
+    Interface to CP Optimizers API
+
+    <some information on the solver>
+
+    Documentation of the solver's own Python API: (all modeling functions)
+    https://ibmdecisionoptimization.github.io/docplex-doc/cp/docplex.cp.modeler.py.html#module-docplex.cp.modeler
+
+    ===============
+    List of classes
+    ===============
+
+    .. autosummary::
+        :nosignatures:
+
+        CPM_template
+"""
+
+class CPM_cpo(SolverInterface):
+    """
+    Interface to CP Optimizers API
+
+    Requires that the 'docplex' python package is installed:
+    $ pip install docplex
+
+    docplex documentation:
+    https://ibmdecisionoptimization.github.io/docplex-doc/
+
+    You will also need to install CPLEX Optimization Studio from IBM's website.
+    There is a free community version available.
+    https://www.ibm.com/products/ilog-cplex-optimization-studio
+
+    See detailed installation instructions at:
+    https://www.ibm.com/docs/en/icos/22.1.2?topic=2212-installing-cplex-optimization-studio
+
+    Creates the following attributes (see parent constructor for more):
+    - cpo_model: object, CP Optimizers model object
+    """
+
+    @staticmethod
+    def supported():
+        return CPM_cpo.installed() and CPM_cpo.executable_installed()
+
+    @staticmethod
+    def installed():
+        # try to import the package
+        try:
+            import docplex.cp as docp
+            return True
+        except ModuleNotFoundError:
+            return False
+
+    @staticmethod
+    def executable_installed():
+        if shutil.which('cpoptimizer') is None:
```

Comment:
I installed it and also the python link, but I did not bother setting up the CLI PATH. So this fails, even though my docplex can model and solve perfectly...

I think this has to be more similar to the gurobi solver interface instaed, where we have a 'license_OK()' function that checks something purely through the python API. If you can't find a way then for all I care it just does docplex.mp.model.Model().solve() or smth...